<!--
  JoyOp launcher

  Includes JoyOp and the ROS joystick driver
 -->

<launch>
  <arg name="device"                default="/dev/input/js0"/>
  <arg name="deadzone"              default="0.05"/>
  <arg name="autorepeat_rate"       default="0"/>
  <arg name="coalesce_interval"     default="0.001"/>
  <arg name="linear_axis"           default="1"/>
  <arg name="angular_axis"          default="0"/>
  <arg name="boost_axis"            default="5"/>
  <arg name="linear_scale"          default="1.0"/>
  <arg name="angular_scale"         default="1.0"/>
  <arg name="boost_scale"           default="2.0"/>
  <arg name="deadman_button"        default="4"/>
  <arg name="enable_button"         default="0"/>
  <arg name="disable_button"        default="1"/>
  <arg name="brake_button"          default="5"/>
  <arg name="spin_frequency"        default="10"/>
  <arg name="joy_topic_name"        default="joy"/>
  <arg name="cmd_vel_topic_name"    default="cmd_vel"/>
  <arg name="brake_vel_topic_name"  default="brake_vel"/>
  <arg name="enable_topic_name"     default="enable"/>
  <arg name="disable_topic_name"    default="disable"/>
  <arg name="wait_for_connection"   default="-1"         doc="time to wait to try and connect/enable via the motor topics on initialisation."/>
  <arg name="enable_on_start"       default="false"/>

  <node pkg="yocs_joyop" type="joyop" name="yocs_joyop" output="screen">
    <param name="linear_axis"           value="$(arg linear_axis)"/>
    <param name="angular_axis"          value="$(arg angular_axis)"/>
    <param name="boost_axis"            value="$(arg boost_axis)"/>
    <param name="linear_scale"          value="$(arg linear_scale)"/>
    <param name="angular_scale"         value="$(arg angular_scale)"/>
    <param name="boost_scale"           value="$(arg boost_scale)"/>
    <param name="deadman_button"        value="$(arg deadman_button)"/>
    <param name="enable_button"         value="$(arg enable_button)"/>
    <param name="disable_button"        value="$(arg disable_button)"/>
    <param name="brake_button"          value="$(arg brake_button)"/>
    <param name="spin_frequency"        value="$(arg spin_frequency)"/>
    <param name="wait_for_connection"   value="$(arg wait_for_connection)"/>
    <param name="enable_on_start"       value="$(arg enable_on_start)"/>
    <remap from="joy"                   to="$(arg joy_topic_name)"/>
    <remap from="yocs_joyop/cmd_vel"    to="$(arg cmd_vel_topic_name)"/>
    <remap from="yocs_joyop/brake_vel"  to="$(arg brake_vel_topic_name)"/>
    <remap from="yocs_joyop/enable"     to="$(arg enable_topic_name)"/>
    <remap from="yocs_joyop/disable"    to="$(arg disable_topic_name)"/>
  </node>

  <!--- ROS joystick driver -->
  <node pkg="joy" type="joy_node" name="joystick" output="screen">
    <param name="dev"               value="$(arg device)"/>
    <param name="deadzone"          value="$(arg deadzone)"/>
    <param name="autorepeat_rate"   value="$(arg autorepeat_rate)"/>
    <param name="coalesce_interval" value="$(arg coalesce_interval)"/>
    <remap from="joy"               to="$(arg joy_topic_name)"/>
  </node>
</launch>
